projects/graph/src/graph_demo.py

```diff
@@ -3,14 +3,65 @@
 """
 Demonstration of Graph and BokehGraph functionality.
 """
+import math
 
 from sys import argv
+from graph import Graph
+from draw import BokehGraph
 
+def getGraph(**kwargs):
+    graph = Graph()  # Instantiate your graph
+    graph.add_vertex(0)
+    graph.add_vertex(1)
+    graph.add_vertex(2)
+    graph.add_vertex(3)
+    graph.add_vertex(4)
+    graph.add_vertex(5)
+    graph.add_vertex(6)
+    graph.add_edge(0, 1)
+    graph.add_edge(1, 2)
+    graph.add_edge(2, 3)
+    graph.add_edge(3, 4)
+    graph.add_edge(4, 5)
+    graph.add_edge(5, 6)
 
-def main():
-    pass  # TODO
+    bokeh_graph = BokehGraph(graph)
+    bokeh_graph.draw()
+
+
+def getRandomGraph(numVerts, numEdges):
+    graph = Graph()
+
+    for vert_id in range(0, numVerts):
+        graph.add_vertex(vert_id)
+
+    allEdges = []
+    for i in range(0, numVerts):
+        for j in range(0, numVerts):
+            if i < j:
+                allEdges.append( (i, j) )
+    randomEdges = allEdges[:numEdges]
+    for edge in randomEdges:
+        graph.add_edge(edge[0], edge[1])
+
+    return graph
 
 
 if __name__ == '__main__':
-    # TODO - parse argv
-    main()
+    style = 'default'
+    num_verts = 5
+    num_edges = 5
+
+    for arg in argv[1:]:
+        arg_split = arg.split("=")
+        if len(arg_split) == 2:
+            if arg_split[0] == "style":
+                style = arg_split[1].lower()
+            elif arg_split[0] == "verts":
+                num_verts = int(arg_split[1])
+            elif arg_split[0] == "edges":
+                num_edges = int(arg_split[1])
+    print(style)
+
+    getGraph(style=style, num_verts=num_verts, num_edges=num_edges)
+    getRandomGraph(5, 2)
```

Comment:
When I run graph_demo, I get the same output in my terminal every time, which are the values you hard coded in lines 12-26. But the colors are different each time. Have you tried commenting out the values and seeing if you can generate random numbers for the graph?